packages/contracts-bedrock/src/L1/SystemConfig.sol

```diff
@@ -90,9 +90,12 @@ contract SystemConfig is OwnableUpgradeable, ISemver, IGasToken {
     uint64 internal constant MAX_GAS_LIMIT = 200_000_000;
 
     /// @notice Fixed L2 gas overhead. Used as part of the L2 fee calculation.
+    ///         Deprecated since the Ecotone network upgrade
     uint256 public overhead;
 
     /// @notice Dynamic L2 gas overhead. Used as part of the L2 fee calculation.
+    ///         The most significant byte is used to determine the version since the
+    ///         Ecotone network upgrade.
     uint256 public scalar;
```

Comment:
Why do we use the most significant byte here instead of adding a new variable?

Reply:
This was done as part of ecotone to enable the upgrade with no diff to the L1 contracts. The semantics of the 32 byte scalar are interpreted differently

Reply:
See https://github.com/ethereum-optimism/specs/blob/362db2981471d3d96871a44985789e838653ee22/specs/protocol/system_config.md#ecotone-scalar-overhead-uint256uint256-change